From d207a03312aed365e91ffc2e1ccbd61a71457caf Mon Sep 17 00:00:00 2001
From: Oskari Rauta <oskari.rauta@gmail.com>
Date: Wed, 3 Nov 2021 03:58:30 +0200
Subject: [PATCH] avahi: do not compile against host's libc

On ArchLinux (and maybe on some other's too) avahi links against host system's libc
instead of target systems (atleast for libdnnsd). This patch fixes the issue and
should have no negative impact when building on other systems.

I also changed release to $(AUTORELEASE)

https://github.com/openwrt/packages/pull/17051

Signed-off-by: Oskari Rauta <oskari.rauta@gmail.com>
---
 libs/avahi/Makefile | 10 ++++++++--
 1 file changed, 8 insertions(+), 2 deletions(-)

diff --git a/libs/avahi/Makefile b/libs/avahi/Makefile
index d8e23cff3724..84130b6ecd2c 100644
--- a/libs/avahi/Makefile
+++ b/libs/avahi/Makefile
@@ -9,7 +9,7 @@ include $(TOPDIR)/rules.mk
 
 PKG_NAME:=avahi
 PKG_VERSION:=0.8
-PKG_RELEASE:=7
+PKG_RELEASE:=$(AUTORELEASE)
 
 PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.gz
 PKG_SOURCE_URL:=https://github.com/lathiat/avahi/releases/download/v$(PKG_VERSION) \
@@ -23,6 +23,7 @@ PKG_CPE_ID:=cpe:/a:avahi:avahi
 
 PKG_INSTALL:=1
 PKG_BUILD_PARALLEL:=1
+PKG_FIXUP:=autoreconf
 
 include $(INCLUDE_DIR)/package.mk
 
@@ -252,6 +253,8 @@ endef
 
 TARGET_CFLAGS += $(FPIC) -DGETTEXT_PACKAGE
 
+CONFIGURE_VARS += LT_SYS_LIBRARY_PATH="$(TOOLCHAIN_DIR)/lib"
+
 CONFIGURE_ARGS += \
 	--enable-shared \
 	--enable-static \
@@ -290,7 +293,10 @@ CONFIGURE_ARGS += \
 	--with-avahi-group=nogroup \
 	--with-avahi-priv-access-group=nogroup \
 	--with-autoipd-user=nobody \
-	--with-autoipd-group=nogroup
+	--with-autoipd-group=nogroup \
+	--with-sysroot=$(TOOLCHAIN_DIR) \
+	--disable-compat-howl \
+	--disable-nls
 
 ifeq ($(BUILD_VARIANT),dbus)
 ifneq ($(CONFIG_PACKAGE_libavahi-compat-libdnssd),)
